is4_fact.loboc

```diff
@@ -0,0 +1,29 @@
+// ISEQ4: Calculates n!
+//tlc type: run
+//tlc case: 1=0
+//tlc case: 1=1
+//tlc case: 2=2
+//tlc case: 6=3
+//tlc case: 24=4
+//tlc case: 125=5
+//tlc case: 720=6
+//tlc case: 5040=7
+
+unsigned n;
+
+{
+    unsigned x;
+    unsigned prod;
+    unsigned[] p;
+
+    x = 1;
+    prod = 1;
+    p = &prod;
+
+    while (x < n) {
+        prod = p[] * x;
+        x++;
+    }
```

Comment:
Out of the box this didn't work for me because of the pointer... what did work was use p directly:
```
unsigned n;

{
    unsigned x;
    unsigned p;
    x = 1;
    p = 1;

    while(x++ < n) {
        p = p * x;
    }

    p;
}
```